src/firefly/js/util/WebUtil.js

```diff
@@ -848,4 +854,32 @@ export function hashCode(str) {
 
 export function isNumeric(n) {
     return !isNaN(parseFloat(n)) && isFinite(n);
-}
\ No newline at end of file
+}
+
+
+export function doDownload (bgStatus, isWorkSpace, wsSelect, fileName) {
```

Comment:
What is the difference between `download` and `doDownload`?  Why is it exported and not used in this PR, but is used in another PR, that's not linked to this one.
If you merge the other PR and not this one, build will fail.  This change should have been with the other PR.  Or, link them.

Reply:
The doDownload is the function I moved from Finderchart to here.   It will be used for workspace download.  In this pull request, it was only used by finderchart.   The download is an existing function.  I just keep it there for now.   Once Firefly-84 is done,  I will make the final clean up.